src/bp/core/services/dialog/dialog-engine.ts

```diff
@@ -36,6 +38,25 @@ export class DialogEngine {
     const currentFlow = this._findFlow(botId, context.currentFlow)
     const currentNode = this._findNode(botId, currentFlow, context.currentNode)
 
+    if (event.ndu) {
+      const workflowName = currentFlow.name?.replace('.flow.json', '')
+
+      const { currentWorkflow, workflows } = event.state.session
+
+      if (currentWorkflow !== workflowName) {
+        this.changeWorkflow(event, workflowName)
+        event.state.session.currentWorkflow = workflowName
+      }
+
+      if (currentNode.type === 'success' || currentNode.type === 'failure') {
```

Comment:
can you extract this condition in a variable, juste to make it clear
`const workflowEnded =currentNode.type === 'success' || currentNode.type === 'failure'`